common/src/main/java/hardcorequesting/common/client/interfaces/graphic/task/CheckBoxTaskGraphic.java

```diff
@@ -0,0 +1,37 @@
+package hardcorequesting.common.client.interfaces.graphic.task;
+
+import hardcorequesting.common.client.ClientChange;
+import hardcorequesting.common.client.interfaces.GuiQuestBook;
+import hardcorequesting.common.client.interfaces.widget.AbstractCheckBox;
+import hardcorequesting.common.network.NetworkManager;
+import hardcorequesting.common.quests.task.CheckBoxTask;
+import net.minecraft.network.chat.TextComponent;
+
+import java.util.UUID;
+
+public class CheckBoxTaskGraphic extends TaskGraphic {
+    private boolean completed;
+
+    public CheckBoxTaskGraphic(CheckBoxTask task, UUID playerId, GuiQuestBook gui) {
+        super(playerId, gui, task);
+
+        completed = task.isCompleted(playerId);
+
+        // TODO custom checkbox label?
+        addClickable(new AbstractCheckBox(gui, TextComponent.EMPTY, START_X, START_Y) {
+            @Override
+            public boolean getValue() {
+                return completed;
+            }
+
+            @Override
+            public void setValue(boolean val) {
+                if (val && !completed) {
+                    task.completeTask(playerId);
+                    completed = true;
+                    NetworkManager.sendToServer(ClientChange.UPDATE_TASK.build(task));
```

Comment:
This sends a call to `task.onUpdate(Player)` which you no longer use. Consider making use of `task.onUpdate()` or remove this packet sending.

Reply:
I thought that would make the task complete on the server-side as well, but this is a dumb way to do it.